[ADD] payroll: adding new contract doc (#4187)
odoo/documentation: content/applications/hr/payroll/contracts.rst

=========
Contracts
=========

Every employee in Odoo is required to have a contract in order to be paid. A contract outlines the
terms of an employee's position, their compensation, their working hours, and any other details
about their position.

.. important::

>    Contract documents (PDF's) are uploaded and organized using the *Documents* application, and are

Reviewer: don't need the apostraphe here
```suggestion
   Contract documents (PDFs) are uploaded and organized using the *Documents* application, and are
```

   signed using the *Sign* application. Ensure these applications are installed in order to send and
   sign contracts. Please refer to the :doc:`/applications/finance/documents` and
   :doc:`/applications/finance/sign` documentation.

To view the employee contracts, go to the :menuselection:`Payroll app --> Employees --> Contracts`
from the top menu.

> .. image:: contracts/payroll-contracts.png

Reviewer: don't need

   :align: center
   :alt: Navigate to the contracts dashboard in Payroll.

All employee contracts, and their current contract status, are displayed in a default kanban view.
The default view displays both running contracts and contracts that need action. Expired and
canceled contracts are hidden in the default view.

.. image:: contracts/contracts-overview.png
   :align: center
   :alt: Contracts dashboard view showing running contracts and contracts with issues.

.. note::
   The list of contracts in the *Payroll* application matches the list of contracts in the
   *Employees* application. The default contracts view in the *Payroll* application displays running
   contracts and contracts needing attention, while the default contracts view in the *Employees*
   application displays all contracts in a kanban view, organized by their stage, regardless of
   status. All contracts can be viewed by changing the filters.

Create a new contract
=====================

In order for an employee to be paid, an active contract is required. If a new contract is needed,
click the :guilabel:`Create` button on the contracts dashboard.

> .. image:: contracts/new-contract-button.png

Reviewer: don't need

   :align: center
   :alt: Click the NEW smart button to create a new contract.

A contract form appears where the information can be entered. Required fields are underlined in
bold.

Required fields
---------------

> .. image:: contracts/required-fields.png

Reviewer: ideally we show a real-life use case on form views like this to better educate the reader (instead of a blank form)

   :align: center
   :alt: New contract form to be filled in when creating a new contract.

- :guilabel:`Contact Reference`: Type in the name or title for the contract, such as `John Smith
  Contract`.
- :guilabel:`Company`: Select the company the contract applies to by clicking on the drop-down menu.
  A new company can be created by typing the name in the field, then clicking either
  :guilabel:`Create` to create the new company, or :guilabel:`Create and Edit` to create the new
  company and edit the company details.
- :guilabel:`Salary Structure Type`: Select one of the salary structure types from the drop-down
  menu. The default salary structure types are :guilabel:`Employee` or :guilabel:`Worker`. A new
  salary structure type can be created by typing the name in the field.

> - :guilabel:`Start Date`: The date the contract starts. Choose a date by clicking on the drop-down
>   menu, navigating to the correct month and year by using the :guilabel:`< > (arrow)` icons, then
>   clicking on the :guilabel:`date`.

Reviewer 2: No need for guilabel on _date_, again this was my previous suggestion :)
```suggestion
- :guilabel:`Start Date`: The date the contract starts. Choose a date by clicking on the drop-down
  menu, navigating to the correct month and year by using the :guilabel:`< > (arrow)` icons, then
  clicking on the date.
```

- :guilabel:`Working Schedule`: Select one of the working schedules from the drop-down menu.

.. tip::
   The :guilabel:`Working Schedule` drop-down menu displays all the working times for the selected
   :guilabel:`Company`. To modify or add to this list, go to :menuselection:`Payroll -->
   Configuration --> Working Times` and either :guilabel:`Create` a new working time or click on an
   existing working time and edit it by clicking :guilabel:`Edit`.

Optional fields
---------------

> .. image:: contracts/optional-fields.png

Reviewer: don't need

- :guilabel:`Employee`: Name of the employee that the contract applies to.
- :guilabel:`Department`: The department the contract applies to.
- :guilabel:`Job Position`: The specific job position the contract applies to.
- :guilabel:`Contract Type`: Chose from :guilabel:`CDI`, :guilabel:`CDD`, or :guilabel:`PFI` from
  the drop-down menu.

   - :guilabel:`CDI` is an open-ended contract with just a start date but no end date.
   - :guilabel:`CDD` is a contract with both a start date and an end date.
   - :guilabel:`PFI` is a Belgian-specific contract used when hiring employees that need training,
     and covers the training period specifically.

- :guilabel:`End Date`: If the contract has a specific end date, enter it here. Chose a date by
  clicking on the drop-down menu, navigating to the correct month and year by using the arrow
  icons, then clicking on the date.
- :guilabel:`HR Responsible`: If there is a specific person in HR that is responsible for the
  contract, select the person from the drop-down menu.
- :guilabel:`Analytic Account`: This field allows a link between the contract and a specific
  analytic account for accounting purposes.

> 

Author: Good idea- I was torn, since if the words were capitalized on the UI I was doing it to match, but since "contract details" is also a generic term, I think it works to keep it sentence case.

   :alt: Contract details in optional tabs for a new contract.

A pop-up window appears with all the contract details. Modify the fields for the contract as needed.

.. image:: contracts/modify-contract.png
   :align: center
   :alt: Edit the details for the contract.

- :guilabel:`Tags`: Select any tags associated with the contract.
- :guilabel:`Signed Document Workspace`: This is where the signatures are stored. Choose a
  pre-configured workspace or create a new one.
- :guilabel:`Signed Document Tags`: Select or create any tags associated only with the signed
  contract as opposed to the original unsigned contract.
- :guilabel:`Redirect Link`: Enter a redirect link for the employee to access the contract. A
  redirect link takes the user from one URL to another, in this case, to the newly updated contract
  specifically written for them.
- :guilabel:`Who can Sign`: Select either :guilabel:`All Users` or :guilabel:`On Invitation`.

  - :guilabel:`All Users`: Any user in the organization can sign the contract.
  - :guilabel:`On Invitation`: Only users selected in this field can sign the contract.

- :guilabel:`Invited Users`: Select the person(s) that can sign the document.

> - :guilabel:`Document`: The attached document can be replaced by clicking the
>   :guilabel:`✏️ (Select)` icon. A pop-up window appears so another document can be selected for
>   upload. The file must be a PDF. To remove the document, click the :guilabel:`🗑️ (Clear)` icon.

Reviewer: early line break here.
also, when referring to emojis, refer to them as what they actually _are_, not what they imply.
e.g. ✏️ is a **pencil** icon not a **select** or **edit** icon.

The purpose of the labels in parenthesis like this is to give readers a literal translation of the icon in case (for whatever reason) the emoji doesn't load on their screen.

```suggestion
- :guilabel:`Document`: The attached document can be replaced by clicking the
  :guilabel:`✏️ (Select)` icon. A pop-up window appears so another document can be selected for
  upload. The file must be a PDF. To remove the document, click the :guilabel:`🗑️ (Clear)` icon.
```

>   must be a PDF. To remove the document, click the clear icon (trash can icon).

Reviewer 3: Maybe you want to modify how these icons are referred to as well (per our writers workshop meeting)?

Author: I added the emojis! I put the proper name that appears when you hover, which is Select and Clear (not intuitive descriptions, honestly).

Once the edits are complete, click the :guilabel:`Save` button. All the information for the selected
contract template populates the fields in the :guilabel:`Salary Information` tab. Any additional
tabs, such as :guilabel:`Personal Documents`, appears if applicable.

Salary information
------------------

.. image:: contracts/salary-info.png
   :align: center
   :alt: Optional tabs for a new contract.

This section is where the specific salary details are defined. This section is country-specific, so
depending on where the company is located, these fields may vary.

Enter the amount in the various fields, or check a box to apply a benefit. Some options that can be
entered here include :guilabel:`Meal Vouchers`, :guilabel:`Fuel Card`, :guilabel:`Internet`,
:guilabel:`Paid Time Off`, etc.

Some fields may be automatically filled in based off of the contracts selected in the
:guilabel:`Contract Details` tab.

Attachment of salary
--------------------

> .. image:: contracts/salary-attachment.png

Reviewer: don't need

   :align: center
   :alt: Optional tabs for a new contract.

Any automatic deductions or allocations for an employee, such as child support payments and wage
garnishments, are referred to as a *salary attachment*. This section is where all of these
deductions or allocations are set.

To add a new deduction, click on :guilabel:`Add a line`. Type in a description for the allocation
under :guilabel:`Description`.

.. image:: contracts/garnishment.png
   :align: center
   :alt: Enter a new line for each type of garnishment.

Select the :guilabel:`Garnished Type` from the drop-down menu. Chose from:

> - :guilabel:`Assignment of Salary` - Any payments taken out towards something that is *not* child

Reviewer 3: The wrong bullet point got changed here. This first one should be "Attachment of Salary".
```suggestion
- :guilabel:`Attachment of Salary` - Any payments taken out towards something that is *not* child
```

  support. Typically any garnishments such as lawsuit payments, payments toward taxes owed, etc.

> - :guilabel:`Allocaiton of Salary` - Any deduction that is not required but voluntary, such as a

Reviewer 3: This second one should be "Assignment of Salary".
```suggestion
- :guilabel:`Assignment of Salary` - Any deduction that is not required but voluntary, such as a
```

  pre-tax allocation to a college savings account.
- :guilabel:`Child Support` - Any payments taken out specifically for child support.

Enter the start and end dates the entry applies to by clicking on the drop-down menu under
:guilabel:`From` and :guilabel:`To`, navigating to the correct month and year by using the arrow
icons, then clicking on the date.

Last, enter the :guilabel:`amount` that each payslip will pay towards the entry.

> Delete a line by clicking on the trash can icon at the end of the line.

Reviewer 3: Maybe modify this icon reference as well?

> .. image:: contracts/delete-garnishment.png

Reviewer: don't need

   :align: center
   :alt: Remove a new line by clicking the delete icon.

Save and send the contract
--------------------------

Once contracts have been created and/or modified, save the contract by clicking the :guilabel:`Save`

> smart button. Next, the contract must be sent to the employee to be signed.

Reviewer 3: This is just a regular Save button, not a smart button.
```suggestion
button. Next, the contract must be sent to the employee to be signed.
```

> Click on one of the following smart buttons:

Reviewer 3: These are regular buttons as well. Also adding a little bit of context to this sentence.
```suggestion
Click on one of the following buttons to send the contract to the employee:
```

.. image:: contracts/send-contract.png
   :align: center
   :alt: Send the contract to the employee via one of the smart buttons.

- :guilabel:`Generate Simulation Link`: This option is for Belgian companies only. Clicking this
  will open a pop-up window that contains the basic information from the contract as well as a link
  for the contract when using the salary configurator. Click :guilabel:`Send Offer` to send an email
  to the employee so they can sign the contract.

.. note::
   In order to send a contract using the :guilabel:`Generate Simulation Link`, there must be a

>    signature field in the contract PDF being sent to the employee so they can sign it.

Reviewer 3: I would move this note to be right below the image that is currently below it (simulation.png). As it is now, I kind of expect the image to illustrate the note since it comes after it.

.. image:: contracts/required-fields.png
   :align: center
   :alt: New contract form to be filled in when creating a new contract.

- :guilabel:`Contact Reference`: Type in the name or title for the contract, such as `John Smith
  Contract`.
- :guilabel:`Company`: Select the company the contract applies to by clicking on the
  :guilabel:`drop-down menu`. A new company can be created by typing the name in the
  :guilabel:`box`, then clicking either :guilabel:`Create` to create the new company, or
  :guilabel:`Create and Edit` to create the new company and edit the company details.
- :guilabel:`Salary Structure Type`: Select one of the salary structure types from the
  :guilabel:`drop-down menu`. The default salary structure types are :guilabel:`Employee` or
  :guilabel:`Worker`. A new salary structure type can be created by typing the name in the field.
- :guilabel:`Start Date`: The date the contract starts. Choose a date by clicking on the
  :guilabel:`drop-down menu`, navigating to the correct month and year by using the arrow icons,
  then clicking on the :guilabel:`date`.
- :guilabel:`Working Schedule`: Select one of the working schedules from the :guilabel:`drop-down

>   menu`.

Author: One small question - should "John Smith" be underlined? It's underlined here, but I wasn't sure if it was supposed to or not.

>   menu`.

Reviewer 2: No underline, that's just GH formatting the text

.. tip::
   The :guilabel:`Working Schedule` drop-down menu displays all the working times for the selected
   :guilabel:`Company`.
   To modify or add to this list, go to :menuselection:`Payroll --> Configuration --> Working Times`
   and either :guilabel:`Create` a new working time or click on an existing working time and
   edit it by clicking :guilabel:`Edit`.

Optional fields
---------------

.. image:: contracts/optional-fields.png
   :align: center
   :alt: Optional fields for a new contract.

- :guilabel:`Employee`: Name of the employee that the contract applies to.
- :guilabel:`Department`: The department the contract applies to.
- :guilabel:`Job Position`: The specific job position the contract applies to.
- :guilabel:`Contract Type`: Choose from :guilabel:`CDI`, :guilabel:`CDD`, or :guilabel:`PFI` from
  the drop-down menu.

   - :guilabel:`CDI` is an open-ended contract with only a start date but no end date.
   - :guilabel:`CDD` is a contract with both a start date and an end date.
   - :guilabel:`PFI` is a Belgian-specific contract used when hiring employees that need training,

>      and covers the training period specifically.

Reviewer 2: Bullet points indented by an extra space

Author: I made this indented one more since these 3 items are explaining the options on the previous bullet.  So since it's linked/not part of the same list, but a sub-list of the list, I indented it.  If that doesn't make sense, I can remove, but I thought the format made sense,

  be entered for future reference.

Modifying a contract
~~~~~~~~~~~~~~~~~~~~

Click the :guilabel:`External Link` button at the end of each line to open the corresponding
contract template and make any changes.

.. image:: contracts/external-link.png
   :align: center
   :alt: Contract details in optional tabs for a new contract.

A pop-up window appears with all the contract details. Modify the fields for the contract as needed.

.. image:: contracts/modify-contract.png
   :align: center
   :alt: Edit the details for the contract.

- :guilabel:`Tags`: Select any tags associated with the contract.
- :guilabel:`Signed Document Workspace`: This is where the signatures are stored. Choose a
  pre-configured workspace or create a new one.
- :guilabel:`Signed Document Tags`: Select or create any tags associated only with the signed
  contract as opposed to the original unsigned contract.
- :guilabel:`Redirect Link`: Enter a redirect link for the employee to access the contract. A
  redirect link takes the user from one URL to another, in this case, to the newly updated contract
  specifically written for them.

> - :guilabel:`Who can Sign`: Select either *All Users* or *On Invitation*.

Reviewer 3: Consider making these guilabels, since you're instructing the reader to select options in the UI. But I think it's okay either way.
```suggestion
- :guilabel:`Who can Sign`: Select either :guilabel:`All Users` or :guilabel:`On Invitation`.
```

   - :guilabel:`All Users`: Any user in the organization can sign the contract.

>    - :guilabel:`On Invitation`: Only users selected in this field can sign the contract.

Reviewer 2: Bullet points indented by an extra space

Author: I did the same thing here- this was an explanation of the 2 options in the bullet above, so I indented it to show that.

- :guilabel:`Invited Users`: Select the person(s) that can sign the document.
- :guilabel:`Document`: The attached document can be replaced by clicking the
  :guilabel:`🖉 (Select)` icon. A pop-up window appears so another document can be selected for

>   upload. The file must be a PDF. To remove the document, click the :guilabel:`🗑️ (Clear)` icon.

Reviewer 2: This icon is not showing up for me, I'm assuming you're trying to use "✏️" which should work. Please try and fix this, otherwise omit the emoji/icon and just use a standard :guilabel:

Also, the :guilabel:s are currently depicting the _role_ of the icon in the parenthesis, rather than the visual depiction within the parenthesis. I think it makes sense this way, but I'm curious on @StraubCreative's thoughts here.

Author: ok- pencil is fixed! But I'll leave this comment open for @StraubCreative =)

   :align: center
   :alt: Optional tabs for a new contract.

This section is where the specific salary details are defined. This section is country-specific, so
depending on where the company is located, these fields may vary.

Enter the amount in the various fields, or check a box to apply a benefit. Some options that can be
entered here include :guilabel:`Meal Vouchers`, :guilabel:`Fuel Card`, :guilabel:`Internet`,
:guilabel:`Paid Time Off`, etc.

Some fields may be automatically filled in based off of the contracts selected in the
:guilabel:`Contract Details` tab.

Attachment of salary
--------------------

.. image:: contracts/salary-attachment.png
   :align: center
   :alt: Optional tabs for a new contract.

Any automatic deductions or allocations for an employee, such as child support payments and wage
garnishments, are referred to as a *salary attachment*. This section is where all of these
deductions or allocations are set.

To add a new deduction, click :guilabel:`Add a line`. Type in a description for the allocation under
:guilabel:`Description`.

.. image:: contracts/garnishment.png
   :align: center
   :alt: Enter a new line for each type of garnishment.

> Select the :guilabel:`Garnished Type` from the drop-down menu. Chose from:

Reviewer 3: Fixing a typo.
```suggestion
Select the :guilabel:`Garnished Type` from the drop-down menu. Choose from:
```

> - :guilabel:`Attachment of Salary` - Any payments taken out towards something that is *not* child

Reviewer 3: The dashes in this bulleted list should all be colons to match the formatting in the rest of the doc.
```suggestion
- :guilabel:`Attachment of Salary`: Any payments taken out towards something that is *not* child
```

  support. Typically any garnishments such as lawsuit payments, payments toward taxes owed, etc.

> - :guilabel:`Assignment of Salary` - Any deduction that is not required but voluntary, such as a

Reviewer 3: Changing the dash into a colon.
```suggestion
- :guilabel:`Assignment of Salary`: Any deduction that is not required but voluntary, such as a
```

  pre-tax allocation to a college savings account.

> - :guilabel:`Child Support` - Any payments taken out specifically for child support.

Reviewer 3: Changing the dash into a colon.
```suggestion
- :guilabel:`Child Support`: Any payments taken out specifically for child support.
```

Enter the start and end dates the entry applies to. Click on the drop-down menu under
:guilabel:`From` and :guilabel:`To`, navigate to the correct month and year by using the arrow

> icons, then click on the date.

Reviewer 2: Please add :guilabel: for the actionable elements in this sentence

> Last, enter the :guilabel:`amount` that each payslip pays towards the entry.

Reviewer 2: Capitalize Amount since it is on the UI

> Enter the start and end dates the entry applies to. Click on the :guilabel:`drop-down menu` under
> :guilabel:`From` and :guilabel:`To`, navigate to the correct month and year by using the arrow
> icons, then click on the :guilabel:`date`.

Reviewer 2: I mixed up the elements that needed :guilabel:s here in my previous suggestion, please use this corrected RST:
```suggestion
Enter the start and end dates the entry applies to. Click on the drop-down menu under
:guilabel:`From` and :guilabel:`To`, navigate to the correct month and year by using the
:guilabel:`< > (arrow)` icons, then click on the :guilabel:`date`.
```

> To delete a line, click the :guilabel:`🗑️ (Trash Can)` icon at the end of the line.

Reviewer 2: Lowercase emoji label
```suggestion
To delete a line, click the :guilabel:`🗑️ (trash can)` icon at the end of the line.
```

Last, enter the *amount* that each payslip will pay towards the entry.

Delete a line by clicking on the *trash can icon* at the end of the line.

.. image:: contracts/delete-garnishment.png
   :align: center
   :alt: Remove a new line by clicking the delete icon.

Save and send the contract
--------------------------

Once contracts have been created and/or modified, save the contract by clicking the :guilabel:`Save`
smart button. Next, the contract must be sent to the employee to be signed.

Click on one of the following smart buttons:

.. image:: contracts/send-contract.png
   :align: center
   :alt: Send the contract to the employee via one of the smart buttons.

- :guilabel:`Generate Simulation Link`: This option is for Belgian companies only. Clicking this
  will open a pop-up that contains the basic information as well as a link for the contract when
  using the slaary configurator. There is a :guilabel:`Send Offer` smart button that will send an

>   email to the employee so they can sign the contract.

Author: Thank you! This was a blind spot- so I dug into this. Apparently the contracts are all stored/sent/etc through the Documents app, and they are signed using the Sign app- so both apps must eb installed. Since the documents ap has written docs for that (so does sign) i mentioned this and linked to those docs since I didn't want to add a whole 'how to create/upload/manage/and sign' docs into this already long doc. But that took me a minute! The sign error appears when there is no signature field for the potential employee to sign in the contract being sent,.

.. note::
   In order to send a contract using the :guilabel:`Generate Simulation Link`, there must be a

>    signature field in the contract PDF being sent to the employee so they can sign it.

Reviewer 2: I think you may want to indent these two elements to be within the bullet point above, since the image and note are specific to that point

- :guilabel:`Signature Request`: Clicking this will have a pop up appear where an email can be typed
  to the employee. Select the document, such as a contract, NDA, or Homeworking Policy, from the
  drop down, and fill out the email section. Click :guilabel:`Send` when the email is ready to be
  sent.

.. image:: contracts/sign-contract.png
   :align: center
   :alt: Request a signature for the contract via email.

> - :guilabel:`Credit Time`: This option is for Belgian companies only. When clicked, a wizard pops up

Reviewer 3: I'm afraid "wizard" might be a little confusing, since we've been using "pop-up window" in the rest of the doc.
```suggestion
- :guilabel:`Credit Time`: This option is for Belgian companies only. When clicked, a pop-up window appears
```

Author: Good point! I used the phrasing Wizard because that's what the PR said happens (I didn't see this happen, so I asked what the process was). A pop-up makes more sense, no matter what it is doing.

>    :alt: Request a signature for the contract via email.

Reviewer 2: Also with this image, I think this should be indented to the bullet point above

Author: I added the space, but it doesn't look any different. Is it because of the "center" part of the code? But the spaces are in the RST file, I just don't see it clearly on the front end. The 'Note' block is clearly indented though, so I know the actual spaces in the RST is correct.
